preview-src/comment.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { useContext, useState, useEffect } from 'react';
+
+import Markdown from './markdown';
+import { Spaced, nbsp } from './space';
+import { Avatar, AuthorLink } from './user';
+import Timestamp from './timestamp';
+import { Comment } from '../src/common/comment';
+import { PullRequest } from './cache';
+import PullRequestContext from './context';
+import { editIcon, deleteIcon } from './icon';
+
+export type Props = Partial<Comment & PullRequest>;
+
+export function CommentView(comment: Props) {
+	const { id, pullRequestReviewId, canEdit, canDelete, user, author, htmlUrl, createdAt, bodyHTML, body, } = comment;
+	const [ bodyMd, setBodyMd ] = useState(body);
+	const { deleteComment, editComment } = useContext(PullRequestContext);
+	const [inEditMode, setEditMode] = useState(false);
+	const [showActionBar, setShowActionBar] = useState(false);
+
+	useEffect(() => {
+		if (body !== bodyMd) {
+			setBodyMd(body);
+		}
+	}, [body]);
+
+	if (inEditMode) {
+		return <EditComment body={body}
+			onCancel={
+				() => setEditMode(false)
+			}
+			onSave={
+				text => {
+					editComment({ comment: comment as Comment, text });
```

Comment:
I think we need to handle failure cases here, completely removing all of the comment text in a review comment should show an error message instead of updating the text

Also, let's update both the actual data and the body markdown - right now if I edit the text, change it, and save it, and click edit again, the original body is shown

Reply:
Fixed.